File: Source code/Java test client/Java test client/src/pptRemoteViewerClient/utils/observers/subjects/ScreenRenewalSubject.java
package pptRemoteViewerClient.utils.observers.subjects;

import java.util.ArrayList;

import pptRemoteViewerClient.utils.observers.ScreenRenewalObserver;

public abstract class ScreenRenewalSubject {
	protected ArrayList<ScreenRenewalObserver> observers = new ArrayList<ScreenRenewalObserver>();
	
	public void addObserver(ScreenRenewalObserver observer) {
		observers.add(observer);
	}
	
	public void removeObserver(ScreenRenewalObserver observer) {
		observers.remove(observer);
	}
	
	public abstract void notify(Object parameter);
}

File: Source code/C#/PPT Remote Viewer Server/Utils/Observers/Subjects/ScreenRenewalSubject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PPTRemoteViewerServer.Utils.Observers.Subjects
{
    public abstract class ScreenRenewalSubject
    {
        protected List<ScreenRenewalObserver> observers = new List<ScreenRenewalObserver>();

        public void AddObserver(ScreenRenewalObserver observer)
        {
            observers.Add(observer);
        }

        public void RemoveObserver(ScreenRenewalObserver observer)
        {
            observers.Remove(observer);
        }

        public abstract void Notify(object parameter);
    }
}
